diff --git a/dots/.config/hypr/hyprland/scripts/fuzzel-emoji.sh b/dots/.config/hypr/hyprland/scripts/fuzzel-emoji.sh
--- a/dots/.config/hypr/hyprland/scripts/fuzzel-emoji.sh
+++ b/dots/.config/hypr/hyprland/scripts/fuzzel-emoji.sh
@@ -6 +6 @@
-emoji="$(sed '1,/^### DATA ###$/d' "$0" | fuzzel --match-mode fzf --dmenu | cut -d ' ' -f 1 | tr -d '\n')"
+emoji="$(sed -n '/^### DATA ###$/,/^### END DATA ###$/ { /^###/d; p; }' "$0" | fuzzel --match-mode fzf --dmenu | cut -d ' ' -f 1 | tr -d '\n')"
@@ -22,6 +22,7 @@
         ;;
 esac
 exit
+: <<'EMOJI_DATA'
 ### DATA ###
 😀 grinning face face smile happy joy :D grin
 😃 grinning face with big eyes face happy joy haha :D :) smile funny
@@ -1951,3 +1952,5 @@
 🛘 landslide rockfall disaster
 🪊 trombone instrument music
 🪎 treasure chest gold loot pirate
+### END DATA ###
+EMOJI_DATA
