--- a/dots/.config/quickshell/ii/modules/ii/bar/Bar.qml
+++ b/dots/.config/quickshell/ii/modules/ii/bar/Bar.qml
@@ -52,7 +52,7 @@
                 }
                 property bool superShow: false
                 property bool mustShow: hoverRegion.containsMouse || superShow
-                exclusionMode: ExclusionMode.Ignore
+                exclusionMode: ExclusionMode.Normal
                 exclusiveZone: (Config?.options.bar.autoHide.enable && (!mustShow || !Config?.options.bar.autoHide.pushWindows)) ? 0 :
                     Appearance.sizes.baseBarHeight + (Config.options.bar.cornerStyle === 1 ? Appearance.sizes.hyprlandGapsOut : 0)
                 WlrLayershell.namespace: "quickshell:bar"
