diff -ruN pa/IGC/AdaptorOCL/dllInterfaceCompute.cpp pb/IGC/AdaptorOCL/dllInterfaceCompute.cpp
--- pa/IGC/AdaptorOCL/dllInterfaceCompute.cpp	2026-08-10 21:56:19.000000000 +0800
+++ pb/IGC/AdaptorOCL/dllInterfaceCompute.cpp	2026-09-12 01:57:23.436094996 +0800
@@ -454,7 +454,7 @@
     GenerateSPIRVExtensionsMD(Context, *LLVMModule, SPIRVBinary.str());
 
     if (IGC_IS_FLAG_ENABLED(ShaderDumpTranslationOnly))
-      LLVMModule->dump();
+      LLVMModule->print(llvm::errs(), nullptr);
   }
 
   return success;
